teuthology/orchestra/remote.py

```diff
@@ -229,20 +229,28 @@ def mkdtemp(self, suffix=None, parentdir=None):
 
         return self.sh(args).strip()
 
-    def mktemp(self, suffix=None, parentdir=None):
+    def mktemp(self, suffix=None, parentdir=None, path=None, data=None,
+               owner=None, mode=None):
         """
         Make a remote temporary file
 
         Returns: the path of the temp file created.
         """
         args = ['mktemp']
-
         if suffix:
             args.append('--suffix=%s' % suffix)
         if parentdir:
             args.append('--tmpdir=%s' % parentdir)
 
-        return self.sh(args).strip()
+        if not path:
+            path = self.sh(args).strip()
+
+        if data:
+            # sudo is set to False since root user can't write files in /tmp
```

Comment:
this is strange clause, root user can write to any directory in fact

Reply:
And btw, if the file is created by other user, than current, you must use sudo, but the whole idea of temporary files is to use the same user you operate.

Reply:
No, it can not anymore. Try writing files using sudo on kernel version post 4.19, I guess. Try ubuntu 20. This is what the entire is about. Please take a look at the commit message/PR description and at the ticket linked.

Reply:
Ah... I see it is because of the new kernel feature, but anyways, why would you explicitly declare False, if it is already a default value?

Reply:
As for https://tracker.ceph.com/issues/49466 this looks like a test bug, not the teuthology. The client code should care about tracking users or just always use default and not use sudo_write_file when it is not needed.

Reply:
Also, if we want to write_file as a specific user, maybe it is a good idea to add user name and group support for _sudo_ argument for write_file.

Reply:
> Ah... I see it is because of the new kernel feature, but anyways, why would you explicitly declare False, if it is already a default value?

So that the reader know it's intentional and required even if the reader isn't aware of context.

Reply:
It is self explanatory, if you create a random file and use write_file api, it is with user without sudo.

Reply:
> It is self explanatory, if you create a random file and use write_file api, it is with user without sudo.

If it was, I wouldn't have written this in the first place - https://github.com/ceph/teuthology/pull/1634#discussion_r603966225. :)

Reply:
I am sorry :-) I don't get it, what can be read differently when we have sudo_write_file and write_file? Why would we have sudo_write_file if write_file would have sudo=True?